enctests/testframework/generatetests.py

```diff
@@ -0,0 +1,149 @@
+# This is for generating tests from the markdown code.
+import os
+import argparse
+import yaml
+import re
+
+try:
+    from yaml import CSafeLoader as SafeLoader
+    from yaml import CSafeDumper as SafeDumper
+
+except ImportError:
+    from yaml import SafeLoader, SafeDumper
+
+def parse_args():
+    parser = argparse.ArgumentParser()
+
+    parser.add_argument(
+        '--root',
+        action='store',
+        default='..',
+        help='Where to look for *.md files to generate the config file.'
+    )
+
+    parser.add_argument(
+        '--output-config',
+        action='store',
+        dest='output_config_file',
+        default="test_configs/documentation_tests.yml",
+        help='Specify output file to write test wrappers to.'
+    )
+
+    args = parser.parse_args()
+
+    return args
+
+def main():
+    args = parse_args()
+    rootdir = os.path.abspath(args.root)
+
+    markdownfiles = []
+    for root, dirs, files in os.walk(rootdir):
+        for name in files:
+            if name.endswith(".md"):
+                markdownfiles.append(os.path.join(root, name))
+
+    allinfo = []
+
+    for markdownfile in markdownfiles:
+        print("------\n", markdownfile)
+        f = open(markdownfile, "r")
+        incomment = False
+        incommand = False
+        command = ""
+        info = ""
+        lastcommandstart = -1
+        linenumber = 0
+        for line in f:
+            linenumber = linenumber + 1
+            if "<!---" in line:
+                incomment = True
+                continue
+            if "-->" in line:
+                incomment = False
+                print("Got comment:", info)
+                continue
+            if "```" in line:
+                if incommand:
+                    incommand = False
+                    if "ffmpeg" in command and info != "":
+                        try:
+                            infostruct = yaml.load(info, SafeLoader)
+                            command = command.replace("\\\n", "").replace("\n", "")
+                            print("Command:", infostruct, command)
+                            allinfo.append({'config': infostruct, 'command': command, 'file': markdownfile, 'line': lastcommandstart})
+                        except:
+                            pass
+                    command = ""
+                    info = ""
+                else:
+                    incommand = True
+                    lastcommandstart = linenumber
+                continue
+            if incomment:
+                info = info + line
+            if incommand:
+                command = command + line
+
```

Comment:
We'll need to close the file here if we don't use the context manager approach.
`f.close()`